python/src/aiconfig/util/parser_utils.py

```diff
@@ -0,0 +1,48 @@
+# imports
+import random
+import time
+ 
+import openai
+# define a retry decorator
+def retry_with_exponential_backoff(
```

Comment:
Please unify with https://github.com/lastmile-ai/lastmile-utils/blob/7e5a977cdb3facbe2c76b0608cf3074144c5af36/python/src/lastmile_utils/lib/core/utils.py#L186 in lastmile-utils so we can use it anywhere. 

The signatures can be combined to get the best of both worlds:
- initial_delay, exponential_base, jitter, max_retries, errors
- logger
- strict pyright type check
- [proposal] thunk only (to force user to decouple the function inputs from the error handling)